### DISCUSSION TITLE
How to show the gas fee + network fee of a contract call on react webapp

### DISCUSSION THREAD

**Discussion by ShahzaibAyyub** [wevm/wagmi#2553] [Q&A]
I want to show the gas fee + network fee (shown on metamask popup) on react app before doing transaction. When I use fetchFeeData(), it only shows me some amount, e.g. metamask shows the approx gasFee will be 0.007ETH but the gasFee fetch from fetchFeeData() only shows 0.000000002 ETH. Here's my code:

```
 const fetchFeeInfo = async () => {
    const feeData = await fetchFeeData();
    const ethValue = utils.formatEther(BigNumber.from(feeData.gasPrice));
    console.log(ethValue);
  };
```

**Comment by SaadNiaziDev** [comment]
fetchFeeData() will show gas fee for ideal transaction. Gas fee varies function to function so different functions have different gas fee.
Etheruem has 14 Gwei base fee, fee will increase if your smart contract code is complex
